bindings/go/src/ucx/request.go

```diff
@@ -64,9 +64,10 @@ func NewRequest(request C.ucs_status_ptr_t, callbackId uint64, immidiateInfo int
 			case UcpAmDataRecvCallback:
 				callback(ucpRequest, requestStatus, uint64(immidiateInfo.(C.size_t)))
 			}
-			if requestStatus != C.UCS_OK {
-				return ucpRequest, NewUcxError(requestStatus)
-			}
+		}
+		if requestStatus != UCS_OK {
+			ucpRequest.Status = requestStatus
```

Comment:
do we need to set status if it's `UCS_OK` or it is a default value?

Reply:
`UCS_OK` is equal to 0, so this is default one that is assigned during the creation

Reply:
ok, but i'd not rely on the fact that UCS_OK is zero
imo, would be cleaner to always initialize status, but it's up to you